### DISCUSSION TITLE
Elements not obeying gravity

### DISCUSSION THREAD

**Discussion by Nick20500** [google-deepmind/mujoco#2260] [Asking for Help]
### Intro

Hi!

I am a undergrad student at ETH Zurich, I use MuJoCo for my research on Multimaterial deformable objects.

### My setup

I use MuJoCo 3.2.5 with python bindings.

### My question

I'm trying to simulate a chair dropping to the ground.
The chair is segmented in its parts its made of and each segment is a flexcomp element element and gets a diffrent youngs modulus assigned to.
In order for the segments of the object to really behave as a one object, I use equality constraints. These are in the ec.xml.

When simulating this scenario, not every part of the chair behaves as it should. As seen here in the [video](https://github.com/user-attachments/assets/52827dee-7511-4ef1-bb45-0a321d976938) one corner seems not to be affected by gravity.

I can't figure out, why such behaviour would occur. Is there an error in my code? Is there a bug in mujoco?

Thank you very much for your time and help.

### Minimal model and/or code that explain my question

The corresponding example with all necesseary files you can find here: 
[mujoco_error_example.zip](https://github.com/user-attachments/files/17975871/mujoco_error_example.zip)

### Confirmations

- [X] I searched the [latest documentation](https://mujoco.readthedocs.io/en/latest/overview.html) thoroughly before posting.
- [X] I searched previous [Issues](https://github.com/google-deepmind/mujoco/issues) and [Discussions](https://github.com/google-deepmind/mujoco/discussions), I am certain this has not been raised before.

**Comment by yuvaltassa** [comment]
@quagla can you take a look?

My guess is that you have strong constraint forces that are somehow "leaking" into the dof forces (i.e. not obeying Newton's 1st law).

**Reply by Nick20500** [reply]
@quagla any progress so far?

**Reply by Nick20500** [reply]
Could the type of equality constraint I use to "glue together" the segments cause this?
Or could the way I set the equality constraint have this effect? The faulty parts in the simulation only appear at the areas where two segments of an object touch and are connected by equality constraints.

**Comment by Nick20500** [comment]
Here are more simulations showing the same faulty behaviour: [example.zip](https://github.com/user-attachments/files/18373183/example.zip). 
@quagla please have a look.

**Comment by yuvaltassa** [comment]
@Nick20500, could you re-render your video with contact force visualization? ('F' key in simulate)

**Reply by Nick20500** [reply]
Here is a screen recording from the mujoco viewer: https://github.com/user-attachments/assets/04b5f086-a254-4b55-946e-5517bbc45ab6

I also could render a coded simulation, then I need to know what tags I must add to the xml to show the contact forces.

**Comment by quagla** [comment]
I would suggest to use pinning instead of constraints if you don't have kinematic loops. It seems to me that the only loop in the chair is the back rest, so you can use a constraint for that one.

**Reply by Nick20500** [reply]
what are kinematic loops?

**Reply by quagla** [reply]
If you think your chair as a graph, you have a loop if you can traverse it by going back to the starting position by crossing each body once.

**Comment by quagla** [comment]
Also, your parts are all colliding with each other, so you have to manually exclude the collision pairs between the parts that are meant to be glued to one another.
